lib/rufo/formatter.rb

```diff
@@ -574,7 +582,20 @@ def visit_string_literal(node)
       # Accumulate heredoc: we'll write it once
       # we find a newline.
       @heredocs << [node, tilde]
-      next_token
+      # Get the next_token while capturing any output.
+      # This is needed so that we can add a comma if one is not already present.
+      captured_output = capture_output { next_token }
+
+      inside_literal_elements_list = !@literal_elements_level.nil? &&
```

Comment:
Should we initialize `@literal_elements_level` in the `initialize` method similar to `@node_level`?

Reply:
Good call. I have fixed this up.